packages/core-data/src/private-selectors.ts

```diff
@@ -240,11 +240,38 @@ export const getTemplateId = createRegistrySelector(
 		// First see if the post/page has an assigned template and fetch it.
 		const currentTemplateSlug = editedEntity.template;
 		if ( currentTemplateSlug ) {
-			const currentTemplate = select( STORE_NAME ).getDefaultTemplateId( {
-				slug: currentTemplateSlug,
-			} );
-			if ( currentTemplate ) {
-				return currentTemplate;
+			const userTemplates = select( STORE_NAME ).getEntityRecords(
+				'postType',
+				'wp_template',
+				{ per_page: -1 }
+			);
+			if ( ! userTemplates ) {
+				return;
+			}
+			const userTemplateWithSlug = userTemplates.find(
+				( { slug } ) => slug === currentTemplateSlug
+			);
+
+			if ( userTemplateWithSlug ) {
+				return userTemplateWithSlug.id;
+			}
+
+			const registeredTemplates = select( STORE_NAME ).getEntityRecords(
+				'postType',
+				'wp_registered_template',
+				{ per_page: -1 }
+			);
+
+			if ( ! registeredTemplates ) {
+				return;
+			}
+
+			const registeredTemplateWithSlug = registeredTemplates.find(
+				( { slug } ) => slug === currentTemplateSlug
+			);
+
+			if ( registeredTemplateWithSlug ) {
+				return registeredTemplateWithSlug.id;
```

Comment:
I'm reverting what I did in https://github.com/WordPress/gutenberg/pull/72049/files#diff-166c46de8d08a1bf4740eae54f4a8fa2296684af9f318ec0baa31f248a9c44be, which was wrong. We shouldn't try to resolve a template for this slug. We should check the list of templates we have, and due to the changes we need to check both registered and user templates. In the long term we should really add a way to query the endpoint by slug.